src/services/dataStore.ts

```diff
@@ -84,14 +84,42 @@ export async function find(pattern: string): Promise<string[]> {
     const _keys = promisify(_redisClient.keys).bind(_redisClient)
     const _storedKeys = await _keys(pattern)
     _redisClient.quit()
-
     return _storedKeys
   } catch (error) {
     logger.error(`Error when finding keys: ${error.message}`)
     throw new Error('Pantry is having critical issues')
   }
 }
 
+export async function scan(
+  cursor: number,
+  pattern?: string,
+  count?: number,
```

Comment:
When using `scan`, `COUNT` defaults to `10` ([docs](https://redis.io/commands/scan)). So if I'm not mistaken, if you don't pass in a count at most you'll only receive 10 accounts.

Reply:
Yeah, this should probably iterate over `scan`s to find the total number of accounts.